include/proposals/referendum_settings.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "proposals_base.hpp"
+
+
+class ReferendumSettings : public Proposal {
+
+  public:
+
+    using Proposal::Proposal;
+
+    void create(std::map<std::string, VariantValue> & args) override;
+
+    void update(std::map<std::string, VariantValue> & args) override;
+
+    void cancel(std::map<std::string, VariantValue> & args) override;
+
+    void evaluate(std::map<std::string, VariantValue> & args) override;
+
+    name get_scope() override;
+
+    name get_fund_type() override;
+
+    void check_can_vote(const name & status, const name & stage) override;
+
+    typedef struct sttg_info 
+    {
+      uint64_t previous_value_uint;
+      double previous_value_double;
+      bool is_float;
+    } 
+    SettingInfo;
+
+  private:
+
+    static constexpr name high_impact = name("high");
+    static constexpr name medium_impact = name("med");
+    static constexpr name low_impact = name("low");
+
+    uint64_t get_required_unity(const name & setting, const bool & is_float);
```

Comment:
shouldn't this return a float?

Reply:
Well, in the case of referendums they were getting the unity and the quorum needed looking at the `config` table to get the values depending on the impact. It is an integer because the `config` table stores integers. Do you think we should move those settings to the `floatconfig` table?

Reply:
No it's fine